code/core/src/telemetry/get-package-manager-info.ts

```diff
@@ -0,0 +1,40 @@
+// eslint-disable-next-line depend/ban-dependencies
+import { execaCommand } from 'execa';
+import { detect } from 'package-manager-detector';
+
+import { getProjectRoot } from '../common';
+
+export const getPackageManagerInfo = async () => {
+  const packageManagerType = await detect({ cwd: getProjectRoot() });
+
+  if (!packageManagerType) {
+    return undefined;
+  }
+
+  let nodeLinker: 'node_modules' | 'pnp' | 'pnpm' | 'isolated' | 'hoisted' = 'node_modules';
+
+  if (packageManagerType.name === 'yarn') {
+    try {
+      const { stdout } = await execaCommand('yarn config get nodeLinker', {
```

Comment:
what happens when you run this in yarn1? I guess if it crashes it will just resort back to node_modules and it's all good?

Reply:
Yes, that's exactly what's happening.